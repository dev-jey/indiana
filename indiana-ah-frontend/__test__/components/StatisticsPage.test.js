import React from 'react';
import { mount } from 'enzyme';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import ConnectedStatisticsPage from '../../src/components/StatisticsPage.jsx';

const initialState = {
  statistics: {
    data: [
      {
        bookmarksStat: 0,
        reactionStat: {
          total: 0,
          like: 0,
          dislike: 0
        },
        commentStat: 0,
        ReadingStat: 0
      }
    ],
    isLoading: false,
    error: ''
  }
};

const mockStore = configureStore([thunk]);
const store = mockStore(initialState);

describe('Profile page component', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <Provider store={store}>
            <ConnectedStatisticsPage />
          </Provider>
        </MemoryRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should unmount', () => {
    const statisticsPage = mount(
      <MemoryRouter>
        <Provider store={store}>
          <ConnectedStatisticsPage />
        </Provider>
      </MemoryRouter>
    );
    statisticsPage.unmount();
    const margin = document.body.style.marginBottom;
    expect(margin).toEqual('30rem');
  });
});
