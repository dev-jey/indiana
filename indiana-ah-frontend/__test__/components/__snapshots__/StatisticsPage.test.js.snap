// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile page component renders correctly 1`] = `
<div
  className="main"
>
  <div
    className="sidenav"
  >
    <a
      aria-current={null}
      className="sidenav__logo-box"
      href="/profile"
      onClick={[Function]}
    >
      <img
        className="sideNavLogos"
        src="test-file-stub"
      />
      <span
        className="sidenav__logo-label"
      >
        Profile
      </span>
    </a>
    <a
      aria-current={null}
      className="sidenav__logo-box"
      href="/dashboard/posts"
      onClick={[Function]}
    >
      <img
        className="sideNavLogos"
        src="test-file-stub"
      />
      <span
        className="sidenav__logo-label"
      >
        Articles
      </span>
    </a>
    <a
      aria-current={null}
      className="sidenav__logo-box"
      href="/statistics"
      onClick={[Function]}
    >
      <img
        className="sideNavLogos"
        src="test-file-stub"
      />
      <span
        className="sidenav__logo-label"
      >
        Statistics
      </span>
    </a>
  </div>
  <section
    className="container-fluid profile statPage"
  >
    <div
      className="mb-7 row"
    >
      <div
        className="col-md-4 col-sm-5 profile-card text-center profile-info mail"
      >
        <div>
          <img
            alt="reading-stat"
            className="statisticsImage"
            src="test-file-stub"
          />
        </div>
        <div
          className="py-2"
        >
          <h3
            className="font-weight-bold"
          >
            You have read 
            <span>
              0
               article(s)
            </span>
          </h3>
        </div>
      </div>
      <div
        className="space col-md-1"
      />
      <div
        className="col-md-7 tabs  profile-card order-1 order-md-4"
      >
        <div
          className="d-flex justify-content-around align-items-center"
        >
          <img
            alt="reaction-stat"
            className="bigImage pt-5"
            src="test-file-stub"
          />
          <div>
            <h3
              className="font-weight-bold"
            >
              You have Reacted 
              <span
                className="px-2"
              >
                0
                 time(s)
              </span>
            </h3>
            <div
              className="d-flex justify-content-around"
            >
              <span>
                <img
                  alt="like-stat"
                  src="test-file-stub"
                />
                <h4 />
              </span>
              <span>
                <img
                  alt="disLike-stat"
                  src="test-file-stub"
                />
                <h4 />
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="col-md-4 profile-card text-center mail order-3 order-md-4"
      >
        <div>
          <img
            alt="bookmark-stat"
            className="statisticsImage"
            src="test-file-stub"
          />
        </div>
        <div
          className="pt-4 font-weight-bold"
        >
          <h3>
            You have bookmarked 
            <span>
              0
               article(s)
            </span>
          </h3>
        </div>
      </div>
      <div
        className="space col-md-1 order-2 order-md-4"
      />
      <div
        className="col-md-7 tabs profile-card order-1 order-md-4"
      >
        <div
          className="d-flex justify-content-around align-items-center"
        >
          <img
            alt="comment-stat"
            className="bigImage pt-5"
            src="test-file-stub"
          />
          <div
            className="mt-5"
          >
            <h3
              className="font-weight-bold"
            >
              You have Commented
            </h3>
            <h1>
              0
               time(s)
            </h1>
          </div>
        </div>
      </div>
    </div>
  </section>
</div>
`;
